src/ioctls/common/kvm_vec.rs

```diff
@@ -0,0 +1,542 @@
+        self.as_original_struct()
+    }
+
+    /// Get a mutable pointer to the KVM struct so it can be passed to the kernel.
+    ///
+    pub fn as_mut_ptr(&mut self) -> *mut T {
+        self.as_mut_original_struct()
+    }
+
+    /// Returns a mut `Vec<KvmArray::Entry>` that contains all the elements.
+    /// It is important to call `mem::forget` after using this vector. Otherwise rust will destroy it.
+    ///
+    unsafe fn as_vec(&mut self) -> Vec<T::Entry> {
+        let entries_ptr = self.as_mut_original_struct().entries_mut().as_mut_ptr();
+        Vec::from_raw_parts(entries_ptr, self.len, self.capacity as usize)
+    }
+
+    /// Get the mutable elements slice so they can be modified before passing to the VCPU.
+    ///
+    /// # Example
+    /// ```
+    /// #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    /// use kvm_ioctls::{CpuId, Kvm, MAX_KVM_CPUID_ENTRIES};
+    ///
+    /// #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    /// fn example() {
+    ///     let kvm = Kvm::new().unwrap();
+    ///     let mut cpuid = kvm.get_supported_cpuid(MAX_KVM_CPUID_ENTRIES).unwrap();
+    ///     let cpuid_entries = cpuid.as_entries_slice();
+    /// }
+    /// ```
+    pub fn as_entries_slice(&self) -> &[T::Entry] {
+        let len = self.as_original_struct().len();
+        unsafe { self.as_original_struct().entries().as_slice(len as usize) }
+    }
+
+    /// Get the mutable elements slice so they can be modified before passing to the VCPU.
+    ///
+    /// # Example
+    /// ```
+    /// #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    /// use kvm_ioctls::{CpuId, Kvm, MAX_KVM_CPUID_ENTRIES};
+    ///
+    /// #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    /// fn example() {
+    ///     let kvm = Kvm::new().unwrap();
+    ///     let mut cpuid = kvm.get_supported_cpuid(MAX_KVM_CPUID_ENTRIES).unwrap();
+    ///     let cpuid_entries = cpuid.as_mut_entries_slice();
+    /// }
+    /// ```
+    ///
+    pub fn as_mut_entries_slice(&mut self) -> &mut [T::Entry] {
+        let len = self.as_original_struct().len();
+        unsafe {
+            self.as_mut_original_struct()
+                .entries_mut()
+                .as_mut_slice(len as usize)
+        }
+    }
+
+    /// Reserves capacity for at least `additional` more `KvmArray::Entry` elements.
+    /// If the capacity is already reserved, this method doesn't do anything
+    ///
+    fn reserve(&mut self, additional: usize) {
+        let desired_capacity = self.len + additional;
+        if desired_capacity <= self.capacity {
+            return;
+        }
+
+        let current_kvm_array_len = self.kvm_array.len();
+        let required_kvm_array_len = KvmVec::<T>::num_elements_to_kvm_array_len(desired_capacity);
+        let additional_kvm_array_len = required_kvm_array_len - current_kvm_array_len;
+
+        self.kvm_array.reserve(additional_kvm_array_len);
+        self.capacity = desired_capacity;
```

Comment:
The capacity should be updated to self.kvm_array.capacity().

Reply:
Done